### PR TITLE
drone: Switch Mac (darwin) pipelines to GitHub Actions

### PR DESCRIPTION
**Pull request by camscale**
Switch the pipelines for the darwin push and tag builds to call GitHub
Actions workflows, similar to what has been done for the Linux
pipelines.

The pipeline `build-darwin-amd64` now incorporates what was done in
the `build-darwin-amd64-pkg`, `build-darwin-amd64-pkg-tsh` and
`build-darwin-amd64-connect` pipelines, so these pipelines have been
removed.

The pipeline `push-build-darwin-amd64` also now calls GHA to perform the
build.

dronegen has been changed so that GHA pipelines emit the workflow inputs
in sorted order rather than random map iteration order, to avoid spurious
changes to `.drone.yml` when a pipeline has multiple inputs.

`.drone.yml` was manually edited to remove the extra pipelines no longer
needed before running `dronegen`.

Issue: https://github.com/gravitational/teleport/issues/20319

### CODE REVIEW

**Comment by camscale** [comment]
I've run push builds on teleport and teleport-private and a tag build on teleport. All working as expected, although I'm not sure how to check the release server for `13.0.0-camh.mac.2` -  i'll download the artifact zip from the action and test the installation of the build artifacts in the morning.

File: dronegen/mac_gha.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package main
```
Comment (r0mant) [inline dronegen/mac_gha.go:1]: Nit: Add license header.

**Review by r0mant — APPROVED** [review]
This may be my favorite PR this year yet 🚀

:shipit:

File: dronegen/mac_gha.go (hunk 1 above)
Comment (camscale) [reply dronegen/mac_gha.go:15]: done

**Review by tcsc — APPROVED** [review]

**Comment by wadells** [comment]
🙌

**Pull request merged**